### PR TITLE
SE_Assignment1

### PR DESCRIPTION
**Pull request by adityareddychowdari**

### CODE REVIEW

**Comment by adityareddychowdari** [comment]
submitted by Aditya reddy(119479) and Gubba gowtham(119467)

File: Image.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		this.width = width;
+		this.height = height;
+		this.pixel = width*height; /*defining  the pixel */ 
+		this.data = new byte[pixel*3]; /* converting pixel in to byte*/
+		}
+
+	public void set(int x, int y, int val) {
+		int RedGreenBlueMax = 0xFFFFFF;
+		int threeColors = RedGreenBlueMax & val;
+		String Str = "0x" + Integer.toOctalString(threeColors);/* converting into OctalString and further dividing into substrings according to the index position*/
```
Comment (floe) [inline Image.java]: Octal doesn't make sense here, you should have used toHexString.

File: Image.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		int c = y*width+x;
+		data[(c)*3] = (byte)Integer.parseInt(Str.substring(2,4),4*4); 
+		data[(c)*3+1] = (byte)Integer.parseInt(Str.substring(4,6),4*4);
+		data[(c)*3+2] = (byte)Integer.parseInt(Str.substring(6,8),4*4);
+		
+		
+	}
+	public void write(String filename) {	
+		try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename))) { 
+			String DataStore = "P6" + this.width + this.height  + 255; /* providing file data store to the image*/
```
Comment (floe) [inline Image.java]: No whitespace between items, header is corrupt.

File: Image.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			
+			writer.write(" ");
+			writer.write(DataStore);
+			
+			for (int row=0; row<this.height;row++) {
+				writer.newLine();
+				for (int col=0;col<this.width;col++) {					
+					writer.write(Integer.toOctalString(Byte.toUnsignedInt((byte)(data[row*this.width+col]))));
```
Comment (floe) [inline Image.java]: Again, no whitespace between individual data items.

File: TestSuite.java [hunk 4]
```diff
@@ -21,7 +21,7 @@
     Image i = new Image(100,100);
     i.set(99,99,0x123456);
     int len = i.data.length;
-    assert (false);
+    assert (true);
```
Comment (floe) [inline TestSuite.java]: Not solved as in ImageTest2.

**Comment by floe** [comment]
Your tests fail (-1), ImageTest3 has not been completed correctly (-1), and your PPM file is invalid (-1). See code for further comments.

**Pull request closed without merging**